### PR TITLE
haywood/ch12881/fix-deeply-nested-array-normalization-in: Initial commit

### PR DESCRIPTION
**Pull request by haywoood**
Issue was with carrying over the index during recursion, it was previously only being applied after the fact.

### CODE REVIEW

File: src/artemis/stores/mapgraph/write.cljs [hunk 1]
```diff
@@ -107,12 +106,14 @@
 
                                   nsed-key (str stub "." (name sel-key))
                                   new-v (if (sequential? v)
-                                          (map #(format-for-cache context (:selection-set sel) %1 nsed-key %2) v (range))
+                                          (mapv (fn [result idx]
+                                                  (format-for-cache context (:selection-set sel) result (str nsed-key "." idx)))
```
Comment (haywoood) [inline src/artemis/stores/mapgraph/write.cljs]: `(str nsed-key "." idx)` is the fix, along with removing the other index tracking.

File: project.clj [hunk 2]
```diff
@@ -49,6 +50,9 @@
               :dependencies
               [[re-frame "0.10.5"]]}
 
+   :stories
+   {:test-paths   ^:replace []}
+
```
Comment (colindresj) [inline project.clj]: Don't think we need this, right?

File: etc/repl/user.clj [hunk 3]
```diff
@@ -0,0 +1,10 @@
+(ns user
+  (:require [figwheel-sidecar.system :as sys]
+            [clojure.pprint :refer [pprint]]
+            [com.stuartsierra.component :as component]))
+
+(def system (component/system-map :figwheel-system (sys/figwheel-system (sys/fetch-config))))
+
+(defn go! []
+  (alter-var-root #'system component/start)
+  (sys/cljs-repl (:figwheel-system system)))
```
Comment (colindresj) [inline etc/repl/user.clj]: I think there's a repl file in the `script` directory. Would be good to keep it as one.

File: project.clj [hunk 4]
```diff
@@ -34,7 +34,8 @@
   :profiles
   {:dev {:dependencies
          [[figwheel-sidecar   "0.5.0"]
-          [binaryage/devtools "0.9.9"]]}
+          [binaryage/devtools "0.9.9"]]
+         :source-paths ["src" "etc/repl"]}
```
Comment (colindresj) [inline project.clj]: Can you add a line above `:source-paths` and break the vector down (just for consistent formatting)

File: src/artemis/stores/mapgraph/write.cljs (hunk 1 above)
Comment (colindresj) [reply src/artemis/stores/mapgraph/write.cljs]: Nice spelunking

File: etc/repl/user.clj (hunk 3 above)
Comment (haywoood) [reply etc/repl/user.clj]: I wish I had asked

File: yarn.lock [hunk 5]
```diff
@@ -1644,6 +1644,10 @@ karma-cljs-test@^0.1.0:
   version "0.1.0"
   resolved "https://registry.yarnpkg.com/karma-cljs-test/-/karma-cljs-test-0.1.0.tgz#cb8605ef0e11f9a6f6d28f56ba5dbdf26f389923"
 
+karma-firefox-launcher@^1.1.0:
+  version "1.1.0"
+  resolved "https://registry.yarnpkg.com/karma-firefox-launcher/-/karma-firefox-launcher-1.1.0.tgz#2c47030452f04531eb7d13d4fc7669630bb93339"
+
```
Comment (colindresj) [inline yarn.lock]: Were you testing on firefox? If so, might want to add it to the `package.json`

**Comment by colindresj** [comment]
Looks all good to me. Would be good to get @yedi 's eyes on this too though. Preemptive 🍔

File: yarn.lock (hunk 5 above)
Comment (haywoood) [reply yarn.lock]: no I just typed `yarn`

Comment (colindresj) [reply yarn.lock]: Weird, no big deal

Comment (colindresj) [reply yarn.lock]: Oh, it must be a transitive dep of karma

**Comment by yedi** [comment]
nice, looks good :hamburger:

**Comment by colindresj** [comment]
:shipit:

**Comment by bballant** [comment]
I have to admit that I don't understand the problem or the fix. But I'm sure it's great!

**Pull request merged**